### PR TITLE
[onert] Add tests for IndexEnumerator

### PR DESCRIPTION
**Pull request by glistening**
It adds test for IndexEnumerator based on
compiler/angor/**/IndexEnumerator.test.cpp.

Signed-off-by: Sanggyu Lee <sg5.lee@samsung.com>

Related: #8935

### CODE REVIEW

**Comment by glistening** [comment]
On x86_64

```
$ Product/x86_64-linux.debug/out/unittest_standalone/misc_index_enumerator_test
Running main() from /home/gyu/z/ONE/externals/GTEST/googletest/src/gtest_main.cc
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from MiscIndexEnumeratorTest
[ RUN      ] MiscIndexEnumeratorTest.iterate_full_range
[       OK ] MiscIndexEnumeratorTest.iterate_full_range (0 ms)
[----------] 1 test from MiscIndexEnumeratorTest (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (0 ms total)
[  PASSED  ] 1 test.
```

**Comment by chunseoklee** [comment]
Can we add a negative test for TCM ?

**Comment by glistening** [comment]
I took a look at `IndexEnumerator`. I have no idea what would be a negative test. It doesn't have much room to add.

File: runtime/libs/misc/CMakeLists.txt [hunk 1]
```diff
@@ -8,4 +10,23 @@ target_link_libraries(nnfw_lib_misc PRIVATE nnfw_common)
...
+endif(NOT ENABLE_TEST)
+
+add_executable(misc_index_enumerator_test src/IndexEnumerator.test.cpp)
+target_link_libraries(misc_index_enumerator_test PRIVATE nnfw_lib_misc)
+
+nnfw_find_package(GTest)
+if(NOT GTest_FOUND)
+    message(STATUS "GTest not avaialble. Skipping lib/misc test build")
+    return()
+endif(NOT GTest_FOUND)
```
Comment (hseok-oh) [inline runtime/libs/misc/CMakeLists.txt:27]: If `GTEST_FOUND` is false, `ENABLE_TEST` become false automatically.
So we don't need this check.

https://github.com/Samsung/ONE/blob/master/infra/nnfw/CMakeLists.txt#L85-L91

**Comment by hseok-oh** [comment]
I suggested to use `*.test.cpp` for unittest. But we cannot filter these naming on code analysis tool.

https://github.com/Samsung/ONE/blob/master/.ahub/sam/exclude.txt

So please use `test/` directory and add exclude list.

**Comment by glistening** [comment]
Hmm.. it is just for cover at most 9 functions. But it seems there are many work to be done to add unit test.
- Negative Test
- Source Relocation for other score

Let me pend this PR. It already took more time than I expected.

**Comment by hseok-oh** [comment]
> I suggested to use `*.test.cpp` for unittest. But we cannot filter these naming on code analysis tool. (it may makes down DC score)
> 
> https://github.com/Samsung/ONE/blob/master/.ahub/sam/exclude.txt
> 
> So please use `test/` directory and add exclude list.

@glistening 
I found out we don't build test code (gtest) on code analysis, so test code doesn't effect to analysis score. 
Sorry for confuse.

**Comment by glistening** [comment]
> @glistening I found out we don't build test code (gtest) on code analysis, so test code doesn't effect to analysis score. Sorry for confuse.

@hseok-oh Never mind. Anyway, it is now under its own test folder as you can see this PR. I will keep this structure. : )

@chunseoklee I've added negative test. Please take a look after all tests pass.

File: .ahub/sam/exclude.txt [hunk 2]
```diff
@@ -24,6 +24,7 @@
 
 # Test codes
 /ONE/tests
+/ONE/runtime/libs/misc/test
```
Comment (glistening) [inline .ahub/sam/exclude.txt:27]: @hseok-oh I dont need to add this according to https://github.com/Samsung/ONE/pull/8971#issuecomment-1103293004. Is it right? Then, I will remove this line.

Comment (hseok-oh) [reply .ahub/sam/exclude.txt:27]: Right. Instead, we need to update `.ahub/tcchecker-tca/config.yaml`.

File: .ahub/tcchecker-tca/config.yaml [hunk 3]
```diff
@@ -17,6 +17,7 @@ test:
       - /runtime/onert/test/ir
       - /runtime/onert/test/util
       - /tests/nnfw_api/src
+      - /runtime/libs/misc/test
```
Comment (glistening) [inline .ahub/tcchecker-tca/config.yaml:20]: @hseok-oh I think it is the only line to be added, but please take a look. It is first time for me to edit this file.

Comment (hseok-oh) [reply .ahub/tcchecker-tca/config.yaml:20]: :+1:

**Review by hseok-oh — APPROVED** [review]
LGTM

**Review by chunseoklee — APPROVED** [review]

**Pull request merged**